src/base/PolymeshTransactionBase.ts

```diff
@@ -451,6 +453,34 @@ export abstract class PolymeshTransactionBase<
     };
   }
 
+  /**
+   * Get the latest processed block from the database
+   *
+   * @note uses the middleware
+   */
+  private async getLatestBlockFromDb(): Promise<BigNumber> {
+    const { context } = this;
+
+    let processedBlock: number;
+    if (context.isMiddlewareV2Enabled()) {
+      ({
+        data: {
+          blocks: {
+            nodes: [{ blockId: processedBlock }],
+          },
+        },
+      } = await context.queryMiddlewareV2<EnsuredV2<QueryV2, 'blocks'>>(latestBlockQuery()));
+    } else {
```

Comment:
This is private, so maybe this is over defensive, but should we check?

On on hand we already have check logic so it might be redundant. Thoughts?
```suggestion
    } else if (context.isMiddlewareEnabled()) {
    ...
    } else {
      throw new Error(...
    }
    
```

Reply:
> On on hand we already have check logic so it might be redundant. Thoughts?

I too think it would be redundant. Also, we would be pretty soon be getting rid of all V1 methods when we will remove all related code.